File: scriptorium_app/docker/Dockerfile.java
# Use a lightweight OpenJDK image
FROM openjdk:11-jdk-slim

# Set the working directory inside the container
WORKDIR /app

# Copy the Java source file into the container and rename it to Main.java
COPY code.java /app/Main.java

# Compile and run the Java program
CMD sh -c "javac /app/Main.java && java -cp /app Main"

File: scriptorium_app/docker/Dockerfile.c
# Use a lightweight GCC image
FROM gcc:latest

# Set the working directory inside the container
WORKDIR /app

# Copy the C source file into the container
COPY code.c /app/code.c

# Compile and run the C program
CMD ["sh", "-c", "gcc /app/code.c -o /app/code && /app/code"]
